kore/app/exec/Main.hs

```diff
@@ -719,20 +714,30 @@ koreRun exeLastModifiedTime LocalOptions{execOptions, simplifierx} = do
 -- file as binary data cannot be displayed on the terminal. We put this functionality in the
 -- kore-exec binary because that's where most of the logic it needed in order to function already
 -- lived.
-koreSerialize :: LocalOptions KoreExecOptions -> Main (KFileLocations, ExitCode)
+koreSerialize ::
+    LocalOptions KoreExecOptions ->
+    Main (KFileLocations, ExitCode)
 koreSerialize LocalOptions{execOptions, simplifierx} = do
-    let KoreExecOptions{definitionFileName} = execOptions
-    let KoreExecOptions{mainModuleName} = execOptions
-    let KoreExecOptions{outputFileName} = execOptions
-    let KoreExecOptions{koreSolverOptions} = execOptions
     serializedDefinition@SerializedDefinition{locations} <-
-        makeSerializedDefinition simplifierx koreSolverOptions definitionFileName mainModuleName
+        makeSerializedDefinition
+            simplifierx
+            koreSolverOptions
+            definitionFileName
+            mainModuleName
     case outputFileName of
         Nothing -> return (locations, ExitFailure 1)
-        Just outputFile -> do
-            compact <- compactWithSharing serializedDefinition & liftIO
-            writeCompact outputFile compact & liftIO
+        Just outputFile -> liftIO $ do
+            execHash <- getExecutablePath >>= Fingerprint.getFileHash
+            compact <- compactWithSharing serializedDefinition
+            withFile outputFile WriteMode $ \outputHandle -> do
+                hPut outputHandle (Binary.encode execHash)
+                hPutCompact outputHandle compact
             return (locations, ExitSuccess)
+  where
+    KoreExecOptions{definitionFileName} = execOptions
+    KoreExecOptions{mainModuleName} = execOptions
+    KoreExecOptions{outputFileName} = execOptions
+    KoreExecOptions{koreSolverOptions} = execOptions
```

Comment:
(nit)
```suggestion
    KoreExecOptions { definitionFileName
                    , mainModuleName
                    , outputFileName
                    , koreSolverOptions
                    } = execOptions
```
(or using `RecordWildCards` and `KoreExecOptions{..}`, but which might not be popular)